extensions-contrib/opentelemetry-emitter/src/test/java/org/apache/druid/emitter/opentelemetry/OpenTelemetryEmitterTest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.emitter.opentelemetry;
+
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import org.apache.druid.java.util.common.DateTimes;
+import org.apache.druid.java.util.emitter.core.Event;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+import org.easymock.EasyMock;
+import org.joda.time.DateTime;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OpenTelemetryEmitterTest
+{
+  private static class NoopExporter implements SpanExporter
+  {
+    public Collection<SpanData> spanData;
+
+    @Override
+    public CompletableResultCode export(Collection<SpanData> collection)
+    {
+      this.spanData = collection;
+      return CompletableResultCode.ofSuccess();
+    }
+
+    @Override
+    public CompletableResultCode flush()
+    {
+      return CompletableResultCode.ofSuccess();
+    }
+
+    @Override
+    public CompletableResultCode shutdown()
+    {
+      return CompletableResultCode.ofSuccess();
+    }
+  }
+
+
+  // Check that we don't call "emitQueryTimeEvent" method for event that is not instance of ServiceMetricEvent
+  @Test
+  public void testNoEmitNotServiceMetric()
+  {
+    final Event notServiceMetricEvent =
+        new Event()
+        {
+          @Override
+          public Map<String, Object> toMap()
+          {
+            return null;
+          }
+
+          @Override
+          public String getFeed()
+          {
+            return null;
+          }
+        };
+
+    OpenTelemetryEmitter emitter = EasyMock.createMock(OpenTelemetryEmitter.class);
+    emitter.emit(notServiceMetricEvent);
+    EasyMock.replay(emitter);
```

Comment:
In my experience most modern Java code uses [Mockito](https://site.mockito.org/) instead of EasyMock.  It looks like most of the core Druid code uses mockito and the use of EasyMock in the codebase is mostly in extensions.